test-packages/native/inst/include/mynative.h

```diff
@@ -2,10 +2,11 @@
 #define H_MYNATIVE_H
 
 #include <Rinternals.h>
+#include <R_ext/Rdynload.h>
 
-SEXP mysum(SEXP vector);
+SEXP Cmysum(SEXP vector);
```

Comment:
Using very similar names makes the test case hard to understand. In the future, consider using qualitatively different names, even if they're just nonsense. Like `mysum`, `mytangent`, etc, or even `blue`, `green`, `red`.